### PR TITLE
[WIP] Compiling the base library on Mac

### PR DESCRIPTION
**Pull request by daniellasry**
- [x] Added .dylib binary names for mac in CMakeLists
- [x] Cleared compiler warnings on mac
- [x] Fixed emulator for mac (`os.ttyname` didn't work for `master`)
- [x] Organized build scripts as:
    - build_mac.sh
    - build_win32.cmd
    - build_win64.cmd
    - build_linux.sh

Mac CI will be added in a separate change. Travis can do it, but dependencies have to be installed with `homebrew` on mac:
https://docs.travis-ci.com/user/multi-os/
https://docs.travis-ci.com/user/installing-dependencies/

### CODE REVIEW

File: emulator/brainflow_emulator/cyton_linux.py [hunk 1]
```diff
@@ -16,10 +16,9 @@ def read (port, num_bytes):
 def get_ports_pty ():
     master, slave = pty.openpty ()
     s_name = os.ttyname (slave)
-    m_name = os.ttyname (master)
-    return master, slave, m_name, s_name
```
Comment (daniellasry) [inline emulator/brainflow_emulator/cyton_linux.py]: ttyname fails on master. but it's not used anyway

File: src/board_controller/openbci/inc/openbci_board.h [hunk 2]
```diff
@@ -25,7 +25,7 @@ class OpenBCIBoard : public Board
     int num_channels;
 
     virtual int open_port ();
-    virtual int send_to_board (char *message);
+    virtual int send_to_board (const char *message);
```
Comment (daniellasry) [inline src/board_controller/openbci/inc/openbci_board.h]: compiler warning when passing string literals here

File: src/board_controller/inc/board.h [hunk 3]
```diff
@@ -9,6 +9,7 @@ class Board
     static std::shared_ptr<spdlog::logger> board_logger;
     static int set_log_level (int level);
 
+    virtual ~Board() {};
```
Comment (daniellasry) [inline src/board_controller/inc/board.h]: compiler warning to add a virtual destructor to abstract class (good idea anyway)

File: emulator/brainflow_emulator/cyton_linux.py (hunk 1 above)
Comment (Andrey1994) [reply emulator/brainflow_emulator/cyton_linux.py]: strange that it fails for master and works for slave I dont see any differences between them, but it doesnt matter for us

File: src/board_controller/inc/board.h (hunk 3 above)
Comment (Andrey1994) [reply src/board_controller/inc/board.h]: There is virtual destructor in OpenBCI class which is a base class for all boards but you are right we should add it here too

**Comment by Andrey1994** [comment]
I don't want to add compiled libraries for Mac
We do it fow Windows and Linux right now but for these platforms you can do it using the same machine via VMs or dualboot. If we add compiled libraries for Mac I will be unable to commit anything(I don't have Mac and probably I will need to ask you to build it otherwise our compiled libraries will not match code). Moreover I wanna remove compiled libraries for Windows and Linux too soon

And one more point - for Windows we are able to freeze minimum supported version using CMAKE_SYSTEM_VERSION for Linux we are able to compile it inside Docker with old glibc(kernel.dll and glibc is forward compatible so it works on all machines)
For Mac there is no CMAKE_SYSTEM_VERSION or Docker Image so we have to build libraries on old MacOS(so if we add compiled libraries for mac manually we support MacOS version >= your current version and as soon as you update your own MacOS supported version will be changed too)

So we have to build MacOS libraries inside CI pipeline and we will be able to freeze minimum supported version there

**Comment by Andrey1994** [comment]
And seems like you've added path to new DLL for Mac only to python, we need to update other bindings too

**Comment by daniellasry** [comment]
I canceled my commit adding the mac binaries.

Added .dylib to matlab and java bindings. Java test passing.

**Comment by daniellasry** [comment]
Hmm, how do I test it?

**Comment by Andrey1994** [comment]
Just run cpp test on Mac(you can copypaste cmd from travis I think)

**Comment by daniellasry** [comment]
it works!

**Comment by Andrey1994** [comment]
Cool!
For future - let's squash commits for such cases, we will keep only working commits in master branch using it

**Comment by daniellasry** [comment]
Okay, will do

**Pull request merged**